commands/poll.js

```diff
@@ -0,0 +1,333 @@
+const { SlashCommandBuilder } = require("@discordjs/builders");
+const {
+  ActionRowBuilder,
+  ButtonBuilder,
+  ButtonStyle,
+  ComponentType,
+  EmbedBuilder,
+} = require("discord.js");
+const dotenv = require("dotenv");
+
+dotenv.config();
+
+const MAX_OPTIONS = 10;
+const OPTION_EMOJIS = [
+  "1️⃣",
+  "2️⃣",
+  "3️⃣",
+  "4️⃣",
+  "5️⃣",
+  "6️⃣",
+  "7️⃣",
+  "8️⃣",
+  "9️⃣",
+  "🔟",
+];
+const CLIENT_ID = process.env.DISCORD_CLIENT_ID;
+const DELAY = 3;
+
+const getTopVotedOption = async (message, interaction) => {
+  const initialResults = await Promise.all(
+    message.reactions.cache.map(async (reaction) => {
+      const emojiName = reaction.emoji.name;
+      const reactionUserIds = await reaction.users.fetch();
+
+      // excludes bot
+      const reactionUsernames = await Promise.all(
+        [...reactionUserIds.keys()]
+          .filter((userId) => userId !== CLIENT_ID)
+          .map(async (userId) => {
+            const user = await interaction.client.users.fetch(userId);
+            if (!user.bot) {
+              return user.username;
+            }
+          })
+      );
+
+      const reactionCount = reactionUsernames.length;
+
+      return {
+        emoji: emojiName,
+        count: reactionCount,
+        reactors: reactionUsernames,
+      };
+    })
+  );
+
+  return initialResults.filter(
+    (item) =>
+      item.count === Math.max(...initialResults.map((result) => result.count))
+  );
+};
+
+const pollSlashCommand = new SlashCommandBuilder()
+  .setName("poll")
+  .setDescription("Create a poll with your own custom options!")
+  .addStringOption((option) =>
+    option
+      .setName("question")
+      .setDescription(
+        "The poll question. You need to put one for this to work!"
+      )
+      .setRequired(true)
+  )
+  .addStringOption((option) =>
+    option
+      .setName("option-1")
+      .setDescription(
+        "The first option of the poll. You need at least two options for this to work!"
+      )
+      .setRequired(true)
+  )
+  .addStringOption((option) =>
+    option
+      .setName("option-2")
+      .setDescription(
+        "The second option of the poll. You need at least two options for this to work!"
+      )
+      .setRequired(true)
+  );
+
+for (let i = 3; i <= MAX_OPTIONS; ++i) {
+  pollSlashCommand.addStringOption((option) =>
+    option
+      .setName(`option-${i}`)
+      .setDescription(
+        i === MAX_OPTIONS
+          ? "This is the last option of your poll!"
+          : `Additional option of the poll. You can add ${
+              MAX_OPTIONS - i
+            } more option/s after this!`
+      )
+      .setRequired(false)
+  );
+}
```

Comment:
Comment: Good call on using a for-loop in adding the other optional options rather than having multiple connecting calls to `SlashCommandBuilder()`. Nice touch as well on setting the descriptions accordingly.